docs/3D_RENDERING.md

```diff
@@ -0,0 +1,92 @@
+
+### Coordinate System
+
+As mentioned, Viv follows the [convention of 3D graphics](https://northstar-www.dartmouth.edu/doc/idl/html_6.2/Coordinates_of_3-D_Graphics.html) with respect to the x, y, and z axes. These axes correspond exactly to the dimensions of the OME model which are labelled on the data loaders that Viv expoerts.
+
+Napari, though, does things differently for very good reasons.
+
+Most commonly, the layout of a `numpy` array for use in [the `python` scientific imaging compute ecosystem](https://scikit-image.org/docs/dev/user_guide/numpy_images.html#coordinate-conventions) is `zyx`:
+
+```python
+import numpy as np
+vol = np.array([np.eye(4), np.eye(4) * 1, np.eye(4) * 2])
+vol.shape
+# (3, 4, 4)
+```
+
+In the microscopy world, for example, the above example would represent 3 z-slice images of `4x4` pixels.
+
+When translating this to volumetric viewing, though, Napari needs to map each (unlabeled) dimension of the array to an axis in 3-space. Furthermore, in order to treat these `numpy` arrays as first-class citizens in the Napari ecosystem, Napari needs to respect the in-memory representation of the `numpy` array (which is generally `C` or row-major):
+
+```python
+vol.flatten()
+# array([1., 0., 0., 0., 0., 1., 0., 0., 0., 0., 1., 0., 0., 0., 0., 1., 1.,
+#        0., 0., 0., 0., 1., 0., 0., 0., 0., 1., 0., 0., 0., 0., 1., 2., 0.,
+#        0., 0., 0., 2., 0., 0., 0., 0., 2., 0., 0., 0., 0., 2.])
+```
+
+However, if Napari wishes to show the volume with the correct orientation while respecting this in-memory represnetation, it is forced to buck the 3D graphics convention. As in the 2D case, Napari actually continues to treat the y-axis going down as positively oriented (as mentioned, the 3D graphics convention for the y-axis runs contrary to the convention in 2D raster graphics, where the y-axis is positively oriented in the downward direction). That is, in Napari, the `[0,0,0]` coordinate of your volumetric data array corresponds to the `(0,0,0)` origin in the 3D visualization space. However, because Viv maintains the graphics convention, the `[0,0,0]` data point (and all other data points) must be anti-diagonally transposed in order to maintain the correct orientation of the volume. That is, if you have a `[k,n,m]` shaped data cube in `numpy`, and you wanted to visualize it in Viv via `Zarr`, for example, the `[0,n,m]` data point is actually at the origin `(0,0,0)` and the `[0,0,0]` data point is at `(0,n,m)`. If Viv did not do this anti-diagonal transposition, the volume would look upside down because the `[0,0,0]` data point would be at `(0,0,0)`, which is the "bottom" of the volume.
```

Comment:
Pretty lost here. I still don't think I understand this anti-diagonal reasoning, but you've thought much more about this than myself.